src/crimson/os/seastore/seastore.cc

```diff
@@ -813,31 +838,50 @@ SeaStore::tm_ret SeaStore::_setattrs(
   logger().debug("{} onode={}",
                 __func__, *onode);
   auto& layout = onode->get_mutable_layout(*ctx.transaction);
-  for (auto& [key, val] : aset) {
+  for (auto it = aset.begin(); it != aset.end();) {
```

Comment:
Iterating throuth all keys is not necessary here I think.
`std::map::find()` would be better and more efficient.